### ISSUE TITLE
something wrong with geometry on middle zooms

### ISSUE THREAD

**Issue by FallenKain** [go-spatial/tegola#399]
Something strange happens with data in tegola (firsts screens by tegola and seconds by tilezen backend)
1) 9/346/160.mvt or something like that - no part of road, duplicating of village name, no part of river
![tegola9](https://user-images.githubusercontent.com/21034331/38809839-9677f9c6-418d-11e8-908b-aebecf40466c.png)
![tilezen9](https://user-images.githubusercontent.com/21034331/38809840-96978778-418d-11e8-9ccc-c67faf4797e1.png)
2) same place still bad - no part of road, no part of river
![tegola10](https://user-images.githubusercontent.com/21034331/38809886-c3c64b76-418d-11e8-91e0-b312ea751352.png)
![tilezen10](https://user-images.githubusercontent.com/21034331/38809887-c3e464e4-418d-11e8-934c-b55d9d95e054.png)

place on map - https://www.openstreetmap.org/#map=11/55.3567/64.4980

**Comment by PetersonGIS** [comment]
I wonder if adding dont_simplify to your config file might resolve this. 

Here's an example:

 [[maps.layers]]
	name = "country_lines"
	provider_layer = "ne.ne_50m_admin_0_boundary_lines_land"
	dont_simplify = true
	min_zoom = 3
 max_zoom = 4

http://tegola.io/documentation/configuration/

**Comment by ARolek** [comment]
Not a perfect reference, but the tegola demo for OSM rendering the [same location](http://demo.tegola.io/?z=10.8049&lat=55.3963&lng=64.4793&style=hot-osm&src=ec2&lib=mapbox)

**Comment by FallenKain** [comment]
version is 0.6.x, try it with 0.6.0 and 0.6.2(now 0.6.2 as backend) with new simplification - same results, on 0.5.0 this holes was too. disable simplification for this and...same(check now with param in layer, earlier checked via global param), don't know why this happens. can execute wkt line of this road(and/or river) for you, if need sure. on yours demo lines are more simplified, maybe coz this?

**Comment by PetersonGIS** [comment]
Is there any data on top of the road and river lines? Or perhaps labels for the road and river lines that are masking the underlying features?

**Comment by FallenKain** [comment]
nope, i add tilezen backend screens for showing this, still tilezen returns me more simplified linestrings in this zooms (not so hard simplified, that's why it's look like linestrings in tegola), maybe it wrong when points of line too close for tile border. Oh didn't wrote that this hole on road becomes smaller and smaller while zoom in, i guess problem of point duplication (village name) has same roots as problem with lines

**Comment by ARolek** [comment]
Can you share the tegola endpoint with this data? I want to inspect the tile edges to see if this is possibly a clipping issue on line strings.

**Comment by FallenKain** [comment]
i think sharing can be a...problem :) can only mvt load i think

**Comment by FallenKain** [comment]
ok we will do virtual machine with some access for you, DB and backend will be there, but need time for it :)

**Comment by FallenKain** [comment]
http://maps01.taxsee.ru/tangram-dev/index.html - render client
http://maps01.taxsee.ru/vector-dev/maps/test_map/{z}/{x}/{y}.mvt - for access to data

**Comment by FallenKain** [comment]
virtual machine versions:
centos 7.4, tegola 0.6.3, postgresql-9.6, postgis-2.4

**Comment by FallenKain** [comment]
using LAYER_SQL:EXECUTE_SQL opts, can execute log if need

**Comment by FallenKain** [comment]
update, i do version with simplified(myself) data and hole on road now very small on zoom 11 (but still exists)

**Comment by ARolek** [comment]
@FallenKain can you expose the TileJSON for the map at http://maps01.taxsee.ru/vector-dev/maps/capabilities/test_map.json? I want to check your map rendering in Mapbox GL to make sure it has nothing to do with Tangram.

**Comment by tierpod** [comment]
First of all, thank you for tegola.

@ARolek try to use this url: maps01.taxsee.ru/vector-dev/capabilities/test_map.json

**Comment by FallenKain** [comment]
that's not tangram for sure, as i show earlier, we have 2 backend with same DB and same render client with different results, that's why it's backend issue for sure

**Comment by ARolek** [comment]
Are the tile endpoints still up? It looks like the capabilities endpoint is but the tiles are now returning 403 Forbidden status codes. 

I might need you to dump the tile with our debug tools so we can take a look at the various stages of the data. @gdey can you provide the steps necessary to dump the offending tile to SVG for review?

**Comment by tierpod** [comment]
Tegola is still up and working, for example this url returns 200 status code:
http://maps01.taxsee.ru/vector-dev/maps/test_map/10/696/321.mvt

**Comment by FallenKain** [comment]
@ARolek 1) screens are with simplification of tegola and not simplified data 2) disabling tegola's simplification returns same result's in map 3) version with simplified data (via postgis functions in DB) returns another result (tegola's simplification enabled) 4) variant with simplified data and disabled tegola's simplification didn't check
still thinking that not simplification trouble, but trouble with math in separation near borders(lines are cuted in tiles, right?), and still point can't be simplified but has problem when near tile border

**Comment by gdey** [comment]
Sorry for the delay in reply:

To get the tool with `go get` (need to have go installed)
```
go get github.com/go-spatial/tegola/cmd/xyz2svg
```

You will find the xyz2svg in your `$GOPATH/bin` directory.

To use it you will need to know the feature id, if you provide the feature id it will generate a svg file for each feature in the z,x,y you give it.

To run the command:
```
xyz2svg --config $path_to_tegola_config -t '$z/$x/$y' -p '$provider.$layer' [-g $feature_id]
```

This will output files into the a directory called `_svg_files`. The svg will have three versions of each of the features, the original, the simplified, and the clipped.

**Comment by ARolek** [comment]
@FallenKain if you can follow @gdey's steps and then send us the SVG we can take a look at the various steps in the geo processing to see if it's obvious when the error is manifesting. If it's not clear, we're going to need to standup the data on this end and recreate the problem.

**Comment by FallenKain** [comment]
https://drive.google.com/open?id=1io4soYuI8Pt-aiLx5lzEnCKCzpMjaWqM

**Comment by FallenKain** [comment]
create 10 zoom for road and river and 9 zoom for place name

**Comment by gdey** [comment]
@FallenKain Thanks, just want to let you know that this is still on my radar. I was able to download the svg files; I'm trying to finish up some code I'm working on right now. Then I can get back to fixing these rendering issues.

**Comment by FallenKain** [comment]
1 more question - is tegola has buffer for tiles or can i somehow to set it in cofig file, maybe this can repair this problem

**Comment by ARolek** [comment]
@FallenKain yes you can configure the tile buffer. This is done as a top level config

```toml
tile_buffer = 32  # set as desired
```

I need to document this. I have an open issue to do so. Planning on doing a round of documentation updates soon.

**Comment by FallenKain** [comment]
hm, good, thx :) also documentaition link on get started page http://tegola.io/documentation/getting-started/ is broken (in the end of 3 step), it's redirect to old url %)

**Comment by ARolek** [comment]
@FallenKain thanks for the docs bug report. Fixing now.

**Comment by FallenKain** [comment]
@gdey @ARolek hi there :) something founded?)

**Comment by ARolek** [comment]
@FallenKain unfortunately not. Progress is slow as we're also juggling a few other things.

**Comment by FallenKain** [comment]
okay(
